src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx

```diff
@@ -0,0 +1,260 @@
+import { useAtom, useAtomValue } from "jotai";
+import React, { useMemo, useState } from "react";
+import { State } from "../../../state";
+import {
+  Button,
+  Flex,
+  Text,
+  HoverCard,
+  Link,
+  Strong,
+  Table,
+  Box,
+  Card,
+  Avatar,
+} from "@radix-ui/themes";
+import { ResetIcon, DownloadIcon } from "@radix-ui/react-icons";
+import SingleTableRow from "./SingleRowInTable";
+import { Profile } from "../../../types";
+import * as schema from "../../../data/schema";
+
+interface Weights {
+  [key: string]: number;
+}
+
+interface TQIEntry {
+  weights: Weights;
+}
+
+interface AdjustmentTableProps {
+  selectedProfile?: Profile[];
+  isProfileApplied: boolean;
+  onResetApplied: () => void; // New prop for reset handler
+}
+
+export const AdjustmentTable: React.FC<AdjustmentTableProps> = ({
+  selectedProfile,
+  isProfileApplied,
+  onResetApplied,
+}) => {
+  const dataset = useAtomValue(State.dataset);
+  if (!dataset) return null;
+
+  const getInitialWeights = (
+    selectedProfile: Profile[] | undefined, // Use the Profile interface here
+    dataset: schema.base.Schema,
+    useDataset: boolean
+  ): { [key: string]: number } => {
+    let weights: Weights = {};
+    if (selectedProfile && selectedProfile.length > 0 && !useDataset) {
+      // Use weights from the first selected profile
+      const profileWeights = selectedProfile[0].importance;
+      weights = { ...profileWeights };
+    } else {
+      Object.entries(dataset.factors.tqi).forEach(([_, tqiEntry]) => {
+        const entry = tqiEntry as TQIEntry;
+        Object.entries(entry.weights).forEach(([aspect, importance]) => {
+          weights[aspect] = importance;
+        });
+      });
+    }
+
+    return weights;
+  };
+
+  // useMemo to calculate initial weights based on selectedProfile or dataset
+  const sliderValues = useMemo(() => {
+    const useDataset = !isProfileApplied;
+    return getInitialWeights(selectedProfile, dataset, useDataset);
+  }, [selectedProfile, dataset, isProfileApplied]);
+
+  const [values, setValues] = useState<{ [key: string]: number }>(
+    () => sliderValues
+  );
```

Comment:
this will go stale i.e. `values` is only ever equal to `sliderValues` the first time you render. If you recalculate sliderValues, `values` won't update. 

I think we still have the same problem as before. There needs to be a single prop representing slider values instead of several, and from the usage of `onResetApplied` it appears that that state shouldn't be in this component. 

What exactly is the overall chain of operations you're trying to model here?

Reply:
the overall flow should be: by default in the configuration "adjust importance" tab, the user could decide whether to use a pre-defined profile or not (`ProfileSelection` should return either the selected profile or the default weights in the initial dataset.), based on the selected profile, `AdjustmentTable` will show sliders whose default values as the profile, and users are able to adjust each slider.

Do you have any suggestions? My previous idea is to get selected profile from `ProfileSelection`,  and `ImportanceAdjustment` passes it `AdjustmentTable` to set it as the initial slider default value. A local state `values` in `AdjustmentTable` is for tracking the sliders adjustment, which is based on the selected profile.

Reply:
pending offline discussion